package.json

```diff
@@ -42,53 +42,108 @@
         "configuration": {
             "title": "Competitive Programming Helper",
             "properties": {
-                "cph.saveLocation": {
+                "cph.general.saveLocation": {
                     "type": "string",
                     "default": "",
                     "description": "Location where generated .tcs and .bin files will be saved. Leave empty to save the file in the source file directory. Use this to clean up your folders."
                 },
-                "cph.timeOut": {
+                "cph.general.timeOut": {
                     "type": "number",
                     "default": 3000,
                     "description": "The time in ms for which a testcase is run before it is killed ( timed-out )."
                 },
-                "cph.argsC": {
+                "cph.language.cArgs": {
                     "title": "Compilation flags for .c files",
                     "type": "string",
                     "default": "",
                     "description": "Space seperated additional flags passed to gcc ( for C) while compiling your file. Example '-O2 -Wall'"
                 },
-                "cph.argsCpp": {
+                "cph.language.cCompiler": {
+                    "type": "string",
```

Comment:
Can you make it `cph.language.cSubmissionCompiler` so it is clear that it's related to Codeforces submission? Similarly in other places.

Reply:
![image](https://user-images.githubusercontent.com/30462443/98294785-86cd5c80-1fd6-11eb-8717-f26a94612ba8.png)

Such a nightmare this sorting thing is lol. With this naming, cpp and c settings are getting jumpled. 
Few other options: `cph.language.c.submissionCompiler` or `cph.language.cCompilerForSubmission`. What should we go for?

Reply:
Alright, in that case, `cSubmissionCompiler` looks good.

Reply:
Did you mean `c.submissionCompiler` (notice the dot)?

Reply:
Yes, `c.submissionCompiler`.